x/vesting/keeper/permissioned_decorator.go

```diff
@@ -0,0 +1,46 @@
+package keeper
+
+import (
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/dymensionxyz/dymension-rdk/x/vesting/types"
+)
+
+// PermissionedVestingDecorator prevents invalid msg types from being executed
+type PermissionedVestingDecorator struct {
```

Comment:
not wired in app.go ante handler

Reply:
the dymension-rdk doesn't actually have antehandler. So I added it here and on rollapp-evm's app.go, we just need to call the Decorator from dymension-rdk